WalletWasabi.Gui/CommandLine/CommandInterpreter.cs

```diff
@@ -56,7 +58,7 @@ public static async Task<bool> ExecuteCommandsAsync(string[] args)
 				return false;
 			}
 
-			return false;
```

Comment:
This is not good. Please don't modify the business logic around here. Last time you did this I had to spend days to fix everything up.

Reply:
Currently (before this PR) the logic is "_if no arguments are passed then run GUI_", but that changes because now even the gui accepts the new `--datadir` option. I don't know if this change is correct or not, but the logic (code) has to change somewhere.

Reply:
> I don't know if this change is correct or not, but the logic (code) has to change somewhere.

I don't know either, I'm just saying it's dangerous, but I won't be able to get away with a code review here and I have to test this PR carefully, how it acts on different platforms from code and from the released binaries.

Reply:
I tested it on Linux and Windows. Here are the outputs:

```
$ wassabee --help
Usage: wassabee [OPTIONS]+
Launches Wasabi Wallet.

  -v, --version              Displays Wasabi version and exit.
  -d, --datadir=VALUE        Directory path where store all the Wasabi data.

Available commands are:

        mix                  Start mixing without the GUI with the specified wallet.
        findpassword         Try to find typos in provided password.
```
```
$ wassabee help
Usage: wassabee [OPTIONS]+
Launches Wasabi Wallet.

  -v, --version              Displays Wasabi version and exit.
  -d, --datadir=VALUE        Directory path where store all the Wasabi data.

Available commands are:

        mix                  Start mixing without the GUI with the specified wallet.
        findpassword         Try to find typos in provided password.
```
```
$ wassabee --version
Wasabi Client Version: 1.1.4
Compatible Coordinator Version: 3
```
```
$ wassabee --unknown
wassabee: Unknown command: --unknown
wassabee: Use `wassabee help` for usage.
```
```
$ wassabee unknown
wassabee: Unknown command: unknown
wassabee: Use `wassabee help` for usage.
```
```
$ wassabee --mix
2019-04-27 17:08:25 INFO 5d154f8b-53ef-4d85-826c-0875617462b2: Wasabi Daemon is starting...
2019-04-27 17:08:25 CRITICAL Daemon: Wallet was not supplied. Add --wallet:WalletName
2019-04-27 17:08:25 INFO 5d154f8b-53ef-4d85-826c-0875617462b2: Wasabi Daemon stopped gracefully.
```
```
$ wassabee --mix --datadir:/home/lontivero/topsecret
2019-04-27 17:08:35 INFO 1cae7037-281b-4d4c-b3d9-493d7c0d134b: Wasabi Daemon is starting...
2019-04-27 17:08:38 CRITICAL Daemon: Wallet was not supplied. Add --wallet:WalletName
2019-04-27 17:08:38 INFO 1cae7037-281b-4d4c-b3d9-493d7c0d134b: Wasabi Daemon stopped gracefully.
```

It also works when all the arguments are correct on both: GUI and Daemon.
Same results on Windows.